interfaces/cython/cantera/composite.py

```diff
@@ -910,64 +979,196 @@ def from_pandas(self, df):
         data = df.to_numpy(dtype=float)
         labels = list(df.columns)
 
-        self.restore_data(data, labels)
+        data_dict = OrderedDict()
+        for i, label in enumerate(labels):
+            data_dict[label] = data[:, i]
+        self.restore_data(data_dict)
 
-    def write_hdf(self, filename, cols=None,
-                  key='df', mode=None, append=None, complevel=None,
-                  *args, **kwargs):
+    def write_hdf(self, filename, *args, cols=None, group=None, name=None,
+                  attrs={}, mode='a', append=False,
+                  compression=None, compression_opts=None, **kwargs):
         """
         Write data specified by *cols* to a HDF container file named *filename*.
         Note that it is possible to write multiple data entries to a single HDF
-        container file, where *key* is used to differentiate data.
-
-        Internally, every HDF data entry is a `pandas.DataFrame` generated by
-        the `to_pandas` method.
-
-        :param filename: name of the HDF container file; typical file extensions
-            are `.hdf`, `.hdf5` or `.h5`.
-        :param cols: A list of any properties of the solution being exported.
-        :param key: Identifier for the group in the container file.
-        :param mode: Mode to open the file {None, 'a', 'w', 'r+}.
-        :param append: If True, a less efficient structure is used that makes
-            HDF entries appendable {None, True, False}.
-        :param complevel: Specifies a compression level for data {None, 0-9}.
-            A value of 0 disables compression.
-
-        Arguments *key*, *mode*, *append*, and *complevel* are mapped to
-        parameters for `pandas.DataFrame.to_hdf`; the choice `None` for *mode*,
-        *append*, and *complevel* results in default values set by pandas.
+        container file, where *group* is used to differentiate data.
+
+        :param filename:
+            Name of the HDF container file; typical file extensions are `.hdf`,
+            `.hdf5` or `.h5`.
+        :param cols:
+            A list of any properties of the solution being exported.
+        :param group:
+            Identifier for the group in the container file. A group may contain
+            multiple `SolutionArray` objects. If 'None', group names default
+            to 'groupN', with N being the number of pre-existing groups within
+            the HDF container file.
+        :param name:
+            Name identifier for a subgroup representing the `SolutionArray`
+            object to be added. If 'None', subgroup names default to 'arrM'
+            with M being the number of pre-existing subgroups within the group.
+        :param attrs:
+            Dictionary of user-defined group attributes.
+        :param mode:
+            Mode to open the file {'a' (default), 'w', 'r+}.
```

Comment:
```suggestion
            Mode to open the file {'a' (default), 'w', 'r+'}.
```
Also, we should probably explain what these modes mean, or at least refer to the documentation of `h5py.File`